pig/src/main/java/org/jboss/pnc/bacon/pig/impl/PigContext.java

```diff
@@ -196,7 +196,7 @@ private static PigContext readContext(boolean clean, Path configDir) {
         String sha = hashDirectory(configDir);
 
         PigContext result;
-        String ctxLocationEnv = System.getenv(PIG_CONTEXT_DIR);
+        String ctxLocationEnv = System.getProperty(PIG_CONTEXT_DIR, System.getenv(PIG_CONTEXT_DIR));
```

Comment:
For reference, for perhaps a future consideration, picocli supports default values of e.g. environment variables ( https://picocli.info/#_variable_interpolation ) so it could be a parameter as well.